backend/formfiller/tools/ai_search_tool.py

```diff
@@ -0,0 +1,198 @@
+    return message, formFields
+
+def _get_search_client() -> Optional[SearchClient]:
+    """Get or lazily initialize the Azure AI Search client.
+    Uses a function attribute for caching to avoid module-level globals.
+    Returns None if not configured.
+    """
+    if hasattr(_get_search_client, "_client"):
+        return getattr(_get_search_client, "_client")  # type: ignore[attr-defined]
+
+    search_endpoint = os.environ.get("AZURE_SEARCH_ENDPOINT")
+    search_key = os.environ.get("AZURE_SEARCH_KEY")
+    index_name = os.environ.get("AZURE_SEARCH_INDEX_NAME")
+
+    client: Optional[SearchClient] = None
+    if search_endpoint and search_key and index_name:
+        credential = AzureKeyCredential(search_key)
+        client = SearchClient(
+            endpoint=search_endpoint,
+            index_name=index_name,
+            credential=credential,
+        )
+    setattr(_get_search_client, "_client", client)  # type: ignore[attr-defined]
+    return client
+
+
+@tool("ai_search_tool")
+def ai_search_tool(query: str) -> str:
+    """
+    Search and retrieve data from the configured Azure AI Search index.
+    Expects the following environment variables to be set:
+    - AZURE_SEARCH_ENDPOINT
+    - AZURE_SEARCH_KEY
+    - AZURE_SEARCH_INDEX_NAME
+    """
+    import logging
+    #import pdb; pdb.set_trace()
+    #query = "water licence application"
+    client = _get_search_client()
+    logging.info(f"ai_search_tool called with query: {query}")
+
+    # Extract message and formFields if query is JSON
+    message, formFields = extract_message_and_formfields(query)
+    logging.info(f"Extracted message: {message}")
+    if formFields:
+        logging.info(f"Extracted formFields: {formFields}")
+    
+    if client is None:
+        return (
+            "Azure Search not configured. Please set AZURE_SEARCH_ENDPOINT, "
+            "AZURE_SEARCH_KEY, and AZURE_SEARCH_INDEX_NAME."
+        )
+
+    try:
+        # Build filter string from formFields
+        filters = []
+        if formFields and isinstance(formFields, list):
+            for field in formFields:
+                # Check if field is a dict with data_id and field_value keys
+                field_id = field.get("data_id")
+                field_value = field.get("fieldValue")
+                
+                if field_id and field_value:
+                    safe_value = str(field_value).replace("'", "''")
+                    filters.append(f"{field_id} eq '{safe_value}'")
+        
+        filter_str = " and ".join(filters) if filters else None
+        if filter_str:
+            logging.info(f"Azure Search filter: {filter_str}")
+            
+        # Prepare search arguments
+        search_args = {
+            "search_text": message,
+            "select": ["*"],
+            "top": 3,
+        }
+        
+        if filter_str:
+            search_args["filter"] = filter_str
+        
+        search_results = client.search(**search_args)
```

Comment:
just trying to understand what sort of search args will occur.
so the search would be something like

```python
search_args = {
            # user message
            "search_text": "Am I eligible for exemption from the water licence fee?",
            # field data_id/fieldValue(s) for current state of form joined together in a string
            "filter": "V1FeeExemptionCategory eq Federal Government and V1FeeExemptionClientNumber eq 123456"
            "select": ["*"],
            "top": 3,
        }
```